airflow/models/taskinstance.py

```diff
@@ -2628,10 +2637,8 @@ def signal_handler(signum, frame):
                 task_orig = self.render_templates(context=context, jinja_env=jinja_env)
 
             if not test_mode:
-                rtif = RenderedTaskInstanceFields(ti=self, render_templates=False)
-                RenderedTaskInstanceFields.write(rtif)
-                RenderedTaskInstanceFields.delete_old_records(self.task_id, self.dag_id)
-
+                rendered_fields = _get_fields(ti=self)
```

Comment:
this line is an important part of this change.
what I'm doing is, i'm separating out the "collecting" of the rendered fields from the "saving" of the rendered fields.  this way they can be collected on the _worker_ side, then sent over the wire where the rpc server just stores them.  previously, collection and storage were all done on the RTIF object itself.